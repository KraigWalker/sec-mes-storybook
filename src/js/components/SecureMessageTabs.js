import { Component } from 'react';
import SecureMessageList from './SecureMessageList/SecureMessageList';
import {
  DRAFT,
  SENT,
  INBOX,
  NEW,
  ARCHIVED as ARCHIVE,
} from '../constants/StringsConstants';
import { TabGroup } from 'web-ui-components/lib/navigation/tab-group';
import { Row, Container } from 'web-ui-components/lib/global/layout';
import { withBreakpoints } from '../components/common/hoc/WithBreakpoint';

export class SecureMessageTabs extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeTab: INBOX,
    };
  }

  onclick = (activeTab) => {
    this.setState({ activeTab });
  };

  render() {
    const { messages, containerSize } = this.props;
    const unreadInboxCount = getInboxUnreadCount(messages);
    const inboxTitle = getInboxTitle(unreadInboxCount);
    const activeTabMessages = getActiveTabMessages(
      this.state.activeTab,
      this.props.messages
    );

    return messages ? (
      <Container
        className={'u-margin-top-2 u-padding-left-0'}
        size={containerSize}
      >
        <TabGroup
          className="u-padding-left-0"
          activeTab={this.state.activeTab}
          onChange={(val) => {
            !this.props.fetching && this.setState({ activeTab: val });
          }}
          tabButtons={[
            {
              title: inboxTitle,
              id: INBOX,
            },
            {
              title: 'Drafts',
              id: DRAFT,
            },
            {
              title: 'Sent',
              id: SENT,
            },
            {
              title: 'Archive',
              id: ARCHIVE,
            },
          ]}
        />
        <Row>
          <SecureMessageList
            messages={activeTabMessages}
            activeTab={this.state.activeTab}
            content={this.props.content}
          />
        </Row>
      </Container>
    ) : (
      <p>Loading...</p>
    );
  }
}

const getInboxUnreadCount = (messages) => {
  if (messages.inboxMessages.length) {
    return messages.inboxMessages.filter((message) => message.status === NEW)
      .length;
  }
  return 0;
};

const getInboxTitle = (messageCount) => {
  return messageCount > 0 ? `Inbox (${messageCount})` : `Inbox`;
};

const getActiveTabMessages = (tab, messages) => {
  switch (tab) {
    case DRAFT:
      return messages.draftMessages;
    case SENT:
      return messages.sentMessages;
    case ARCHIVE:
      return messages.archivedMessages;
    case INBOX:
    default:
      return messages.inboxMessages;
  }
};

export default withBreakpoints(SecureMessageTabs);
