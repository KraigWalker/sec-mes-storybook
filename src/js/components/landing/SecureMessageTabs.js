import React from 'react';
import { Tabs, Tab, Grid, Row, Col } from 'react-bootstrap';
import SecureMessageList from './SecureMessageList';
let titleName = ['Inbox', 'Drafts', 'Sent'];

class SecureMessageTabs extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
        const { messages } = this.props.messages;
		return (messages?<div>
				<Tabs activeKey={this.props.activeTab} onSelect={this.props.onClick} id="secure_tabs" className="c-scroll-tabs">
					<Tab eventKey={'inbox'} title={<div>{titleName[0]}</div>}>
                        <SecureMessageList messages={messages? messages.inboxMessages : null}/>
					</Tab>
					<Tab eventKey={'drafts'} title={<div>{titleName[1]}</div>}>
                        <SecureMessageList messages={messages? messages.draftMessages : null}/>
					</Tab>
					<Tab eventKey={'sent'} title={<div>{titleName[2]}</div>}>
                        <SecureMessageList messages={messages? messages.sentMessages : null}/>
					</Tab>
				</Tabs>
			</div> : <p>Loading...</p> 
		);
	}
}

SecureMessageTabs.propTypes = {
	
};
SecureMessageTabs.defaultProps = {
	
};

export default SecureMessageTabs;