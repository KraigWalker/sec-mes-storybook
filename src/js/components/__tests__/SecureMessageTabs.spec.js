import React from 'react';
import {SecureMessageTabs} from '../SecureMessageTabs';
import {shallow} from 'enzyme';
import {TabButton} from ""

describe("New Secure message snapshot", () => {
    const dispatch = jest.fn();
    let props = {
        content: {
            back: 'Back',
        },
        messages: {
            inboxMessages: [
                {
                    status: 'NEW'
                },
            ]
        },
        dispatch: dispatch
    };
    let component = shallow(<SecureMessageTabs {...props} />);
    const tabs = [{id: "INBOX", title: "Inbox (1)"}, {id: "DRAFT", title: "Drafts"}, {
        id: "SENT",
        title: "Sent"
    }, {id: "ARCHIVED", title: "Archive"}];
    it('should match to snapshot', () => {
        expect(component).toMatchSnapshot();

    });
    it("renders the tab groups", () => {
        const {tabButtons} = component.find("TabGroup").props();
        expect(Array.isArray(tabButtons)).toBe(true);
        expect(tabButtons).toEqual(tabs);
    });
    it('onClick method test', () => {
        component.instance().onclick();
        component.setState({tab: 'inbox'});
        expect(component.instance().state.tab).toBe('inbox');
    });
});
