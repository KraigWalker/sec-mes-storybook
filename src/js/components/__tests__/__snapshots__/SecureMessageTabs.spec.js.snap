// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Secure message snapshot should match to snapshot 1`] = `ShallowWrapper {}`;
